### PR TITLE
external/webserver Support of Webserver Handling persitent connections, chunk transfer

### PR DESCRIPTION
**Pull request by amitpurwar**
Support of  Handling persitent connections in webserver, Supporting chunk transfer file in webserver,  Adding a new api for sending response message,  Fix null pointer exception for webserver,  Fix print format correctly

### CODE REVIEW

File: external/webserver/http_client.c
Comment (abhishek-samsung) [inline external/webserver/http_client.c:267]: Please leave spaces as per the coding rules. "if(response){" --> "if (response) {"

File: external/webserver/http_client.c
Comment (abhishek-samsung) [inline external/webserver/http_client.c:322]: Please leave spaces as per the coding rules. "if(response){" --> "if (response) {"

File: external/webserver/http_client.c
Comment (abhishek-samsung) [inline external/webserver/http_client.c:370]: Please leave spaces as per the coding rules. "if(response){" --> "if (response) {"

File: external/webserver/http_client.c
Comment (abhishek-samsung) [inline external/webserver/http_client.c:560]: please change  receied -->  received

File: external/webserver/http_client.h
Comment (pillip8282) [inline external/webserver/http_client.h:78]: how about to add description of the function because it has too many parameters.
I think description will help to understand the function

File: external/webserver/http_client.c
Comment (pillip8282) [inline external/webserver/http_client.c:903]: it seems that but is not initialized after malloc. I'm not sure does snprintf append null character to end of string. If not then you should initialize buf.

File: external/webserver/http_client.c
Comment (pillip8282) [inline external/webserver/http_client.c:837]: If possible you can implement this function with curl. then I think you can implement below features easier.

**Review by pillip8282 — COMMENTED** [review]
http_send_response and http_send_response_with_status_msg those two function has common part. If so how about to remove duplicate part.

File: external/webserver/http_client.c
Comment (sunghan-chang) [reply external/webserver/http_client.c:560]: begining -> beginning

**Comment by sunghan-chang** [comment]
There are build errors. They should be resolved before merging.

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:267]: done

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:322]: done

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:370]: done

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:560]: done

File: external/webserver/http_client.h
Comment (amitpurwar) [reply external/webserver/http_client.h:78]: done

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:903]: done

File: external/webserver/http_client.c
Comment (amitpurwar) [reply external/webserver/http_client.c:837]: ...

**Review by pillip8282 — APPROVED** [review]
LGTM
Please check CI result. I'll merge it after fixing it

**Review by pillip8282 — APPROVED** [review]

**Pull request merged**
